apps/checkout-app/types/api.ts

```diff
@@ -9,10 +9,17 @@ import {
   PaymentProviderSettingID,
   PrivateSettingID,
   PublicSettingID,
-  SettingID,
   SettingsType,
 } from "./common";
 
+export interface Encrypted<T> {
+  encrypted: T;
+}
+export type EncryptionType = "encrypted" | "unencrypted";
+export type SettingsField<E extends EncryptionType> = E extends "unencrypted"
+  ? string | undefined
+  : string | Encrypted<string>;
```

Comment:
Not a fan of this data structure, I'd propose this:

export interface SettingValue<T> {
  value: T;
  encrypted: boolean;
}

to have the exact shape for encrypted and nonecrypted, this data structure will fail when a public value of `"encrypted"` will be used